dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2130,171 @@ def _update_chi2(self):
+        :return: merged null replication metrics
+        :rtype: dict
+        """
+        get_data_type = lambda profile: profile.profiles[
+            "data_type_profile"
+        ].selected_data_type
+        get_data_type_profiler = lambda profile: profile.profiles[
+            "data_type_profile"
+        ]._profiles[get_data_type(profile)]
+
+        self_row_sum = np.asarray(
+            [get_data_type_profiler(profile).sum for profile in self._profile]
+        )
+        other_row_sum = np.asarray(
+            [get_data_type_profiler(profile).sum for profile in other._profile]
+        )
+        total_row_sum = self_row_sum + other_row_sum
+        merged_properties = defaultdict(dict)
+        for col_id in range(len(self._profile)):
+            self_profile = self._profile[col_id]
+            other_profile = other._profile[col_id]
+
+            self_null_count = getattr(self_profile, "null_count")
+            other_null_count = getattr(other_profile, "null_count")
+            null_count = self_null_count + other_null_count
+            if null_count == 0:
+                continue
+
+            self_sample_size = getattr(self_profile, "sample_size")
+            other_sample_size = getattr(other_profile, "sample_size")
+            sample_size = self_sample_size + other_sample_size
+            true_count = sample_size - null_count
+
+            prior_null = null_count / sample_size
+            prior_not_null = true_count / sample_size
+
+            self_sum_null = (
+                self._null_replication_metrics[col_id]["class_sum"][1]
+                if col_id in self._null_replication_metrics
+                else None
+            )
+            other_sum_null = (
+                other._null_replication_metrics[col_id]["class_sum"][1]
+                if col_id in other._null_replication_metrics
+                else None
+            )
+            # Initialize zeros array of size (number of columns - 1)
+            sum_null = np.zeros(len(self._profile) - 1)
+
+            # Add sum_nulls if they exist
+            # if null_count == 0 guarantees that at least one of self_sum_null, other_sum_null is not None
+            if self_sum_null is not None:
+                sum_null += np.asarray(self_sum_null)
+
+            if other_sum_null is not None:
+                sum_null += np.asarray(other_sum_null)
+
+            sum_not_null = np.delete(total_row_sum, col_id) - sum_null
+
+            mean_null = sum_null / null_count
+            mean_not_null = sum_not_null / true_count
+
+            # Convert numpy arrays to lists (serializable)
+            sum_null = sum_null.tolist()
+            sum_not_null = sum_not_null.tolist()
+
+            mean_null = mean_null.tolist()
+            mean_not_null = mean_not_null.tolist()
+
+            merged_properties[col_id] = {
+                # Array index serves as class label
+                # 0 indicates not null, 1 indicates null
```

Comment:
nit picky and not going to hold the PR up over it... just move the comments to above the dict and not inside the dict.